checker/src/main/java/org/checkerframework/checker/signature/qual/GenericSignatureWithTypeParameters.java

```diff
@@ -0,0 +1,16 @@
+package org.checkerframework.checker.signature.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * A signature that represents a class or method or declaration.
+ *
+ * @see GenericSignature
+ * @see MethodSignature
+ * @see ClassSignature
+ */
+@SubtypeOf(GenericSignature.class)
+@Target({ElementType.TYPE_USE, ElementType.TYPE_PARAMETER})
+public @interface GenericSignatureWithTypeParameters {}
```

Comment:
Since this represents the signature for a declaration, I think it would be simpler and clearer to give it a name that reflects that fact, such as `GenericSignatureForDeclaration` or even `SignatureForDeclaration`.